File: examples/list_people.py
#! /usr/bin/env python

# See README.txt for information and build instructions.

from __future__ import print_function
from google import protobuf
protos = protobuf.protos("addressbook.proto")
import sys


# Iterates though all people in the AddressBook and prints info about them.
def ListPeople(address_book):
  for person in address_book.people:
    print("Person ID:", person.id)
    print("  Name:", person.name)
    if person.email != "":
      print("  E-mail address:", person.email)

    for phone_number in person.phones:
      if phone_number.type == protos.Person.MOBILE:
        print("  Mobile phone #:", end=" ")
      elif phone_number.type == protos.Person.HOME:
        print("  Home phone #:", end=" ")
      elif phone_number.type == protos.Person.WORK:
        print("  Work phone #:", end=" ")
      print(phone_number.number)


# Main procedure:  Reads the entire address book from a file and prints all
#   the information inside.
if len(sys.argv) != 2:
  print("Usage:", sys.argv[0], "ADDRESS_BOOK_FILE")
  sys.exit(-1)

address_book = protos.AddressBook()

# Read the existing address book.
with open(sys.argv[1], "rb") as f:
  address_book.ParseFromString(f.read())

ListPeople(address_book)

File: examples/list_people.rb
#! /usr/bin/env ruby

require './addressbook_pb'
require 'pry'

# Iterates though all people in the AddressBook and prints info about them.
def list_people(address_book)
  address_book.people.each do |person|
    puts "Person ID: #{person.id}"
    puts "  Name: #{person.name}"
    if person.email != ""
      puts "  Email: #{person.email}"
    end

    person.phones.each do |phone_number|
      type =
        case phone_number.type
        when :MOBILE
          "Mobile phone"
        when :HOME
          "Home phone"
        when :WORK
          "Work phone"
        end
      puts "  #{type} #: #{phone_number.number}"
    end
  end
end

# Main procedure:  Reads the entire address book from a file and prints all
#   the information inside.
if ARGV.length != 1
  puts "Usage: #{$PROGRAM_NAME} ADDRESS_BOOK_FILE"
  exit(-1)
end

# Read the existing address book.
f = File.open(ARGV[0], "rb")
address_book = Tutorial::AddressBook.decode(f.read)
f.close

list_people(address_book)
